src/idl_gen_lua.cpp

```diff
@@ -0,0 +1,730 @@
+/*
+ * Copyright 2014 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ // independent from idl_parser, since this code is not needed for most clients
+
+#include <string>
+
+#include "flatbuffers/code_generators.h"
+#include "flatbuffers/flatbuffers.h"
+#include "flatbuffers/idl.h"
+#include "flatbuffers/util.h"
+
+#include <unordered_set>
+
+namespace flatbuffers {
+namespace lua {
+
+  // Hardcode spaces per indentation.
+  const std::string Indent = "    ";
```

Comment:
`const char *` ? prefer to not do memory allocations at the static level

Reply:
Ah, I looked this up. Good info, i'll fix it.

Reply:
This added some complications to the `std:string code +=` lines. If the lines just concat `const char *` it throws a compiler error since the rvalue is just adding two pointers and is not automatically cast to `std::string`.

```
static const char * Indent = "    ";
///
std::string code;
code += Indent + Indent; // compiler warning
````

so the code needs to be changed to:

```
static const char * Indent = "    ";
///
std::string code;
code += std::string(Indent) + Indent;
````

or a more major refactoring to fix the issue.

Reply:
Yes, constructing the `std::string` where you are using it is still better than global.